### ISSUE TITLE
Complimentary Escape Hatch in Chamber 13

### ISSUE THREAD

**Issue by Deconimus** [mwpenny/portal64-still-alive#10]
#### Version of the game
Master Branch, commit `d99af47`

#### Original Hardware or Emulator?
Project64 3.0.1

#### What is the bug?
The "complimentary escape hatch" in chamber 13 along with GlaDOS' dialog is triggered falsely.

#### Steps to reproduce the behavior:
1. In test chamber 13
2. Place two portals such that you can reach the cube in the first room.
3. Put cube on the button, such that the first door opens.
4. Shoot a portal through the door at any portal-able wall, so that you can still reach the other portal in your room.
5. Pick up the cube from the button.
6. Go through the portal in your room while holding the cube.
7. GlaDOS dialog is triggered and the exit door opens.

#### Additional context
This does not match the original game's behavior. AFAIK this only happened if you managed to bring at least one cube over to the exit such that the exit door is shut again, trapping the player if they'd remain in the chamber. In the original game, this happened regardless of the player still having access to the "locked out" cube through the portals (can be observed in [this video I've found](https://youtu.be/C6HxZAYWx4E?t=307)).

**Comment by mwpenny** [comment]
Thanks for the detailed description, and nice catch!

This isn't too surprising, since chambers 13 and 14 didn't get the same amount of polish as the others before James stopped working on the project.

The problem lies with the conditions that define whether you are trapped: https://github.com/mwpenny/portal64-still-alive/blob/c90b8ba42871d0f951ef2ed62e85b96b787791cf/assets/test_chambers/test_chamber_08/test_chamber_08.yaml#L26-L32

Basically, you are considered trapped if:
1. You are not in the same room as a cube, and
2. Neither cube is in the first two rooms, or the door to the first room is closed (so even if a cube was there, you couldn't get it)

This assumes you will only be trapped when in the main room. It breaks down in the case where you are in the first room with no cubes, which is what happens when slowly walking through the portal in step 6. The `CHECK_TRAPPED` cutscene requires the `trapped` signal to be set for 30 frames, so that's why you won't see this if you move through the portal quickly.

I also found that it will happen through completing the level normally. The end corridor is outside of the "main room" trigger and so once you get there, you will again be considered trapped.

The trapped conditions just need to be reworked here. Right now, before making level changes, my focus has been learning and [documenting](https://github.com/mwpenny/portal64-still-alive/blob/master/documentation/levels/README.md) how they work. I'm almost done this and just need to cover cutscenes.

I can take a closer look at this after I'm finished documenting cutscenes. In the mean time, this is a good first issue for anyone who wants to get familiar with level editing. Specifically, my pages on [signals](https://github.com/mwpenny/portal64-still-alive/blob/master/documentation/levels/signals.md) and [triggers](https://github.com/mwpenny/portal64-still-alive/blob/master/documentation/levels/level_objects/trigger.md) are relevant here.

**Comment by Deconimus** [comment]
Thank you for the detailed explanation! After reading your response I did some more testing with the original game and was curious whether I could get the same behavior editing the operators in the YAML file. I went ahead and created a pull request:

https://github.com/mwpenny/portal64-still-alive/pull/12

**Explanation:**

* If the player is inside the side_room without a cube, they can still stand on the button and shoot a portal through the door. `trapped` is therefore never reached if the player is inside the side_room.
* The old `main_cube_access` is redundant, as the main room can always be accessed as long as the player is not inside the exit corridor with the exit door closed. This also fixes the cutscene from triggering if the player exits the chamber normally as you described.
* Trapping the first cube in the side_room without the side_room door being open doesn't trigger the cutscene anymore, just like in the original game. The player can still stand on one of the buttons.
* Trapping both cubes in the side_room without the door being open and the player being in the main_room does trigger the cutscene as it should.
* Trapping both cubes in the exit corridor triggers the cutscene as it should.

I really appreciate that this project is being continued btw. I only became aware of its existence after the original repository was already shut down.
